serverless/serverless
# Increase the specificity of the logging rights.

## What did you implement:
Implement a least-privilege logging rights statement in the default IAM policy.



## How did you implement it:

After a number of iterations on this PR, narrowing the logging rights of the deployed lambdas has boiled down to:

1. Remove the log group creation IAM right from the default Lambda execution policy - create the log group using the CFT instead of expecting the Lambda(s) to implicitly create it/them.
2. Rather than have a wild-card-style declaration of the resources that the lambdas are allowed to write into, specify the exact LogGroups that any of the lambdas will be writing into as an array of resources.



## How can we verify it:

Added tests to verify and confirm correct addition of the LogGroup(s)
Add tests to verify and confirm correct addition/specification of LogGroup(s) as the Resource(s) to which the Lambdas have CreateLogStream and PutLogEvents rights upon.

Beyond this programmatic verification, full verification will require deployment of a service and the verification that:

1. logging still works from the lambda's console.log statements into CloudWatch
2. that functions cannot `PutLogEvents` (say, via explicit coding against the CloudWatch API) into any CloudWatch logs not listed in the policy

Apologies but please note that I cannot deploy policies and as such cannot easily verify these things myself.



## Todos:

- [x] Write documentation // this change doesn't impact docs, it simply increases rights specificity
